core/block_flyout_inflater.ts

```diff
@@ -0,0 +1,200 @@
+import type {IFlyout} from './interfaces/i_flyout.js';
+import type {IFlyoutInflater} from './interfaces/i_flyout_inflater.js';
+import type {IBoundedElement} from './interfaces/i_bounded_element.js';
+import {BlockSvg} from './block_svg.js';
+import type {WorkspaceSvg} from './workspace_svg.js';
+import * as utilsXml from './utils/xml.js';
+import * as eventUtils from './events/utils.js';
+import * as Xml from './xml.js';
+import * as blocks from './serialization/blocks.js';
+import * as common from './common.js';
+import * as registry from './registry.js';
+import {MANUALLY_DISABLED} from './constants.js';
+import type {Abstract as AbstractEvent} from './events/events_abstract.js';
+import type {BlockInfo} from './utils/toolbox.js';
+import * as browserEvents from './browser_events.js';
+
+/**
+ * The language-neutral ID for when the reason why a block is disabled is
+ * because the workspace is at block capacity.
+ */
+const WORKSPACE_AT_BLOCK_CAPACITY_DISABLED_REASON =
+  'WORKSPACE_AT_BLOCK_CAPACITY';
+
+export class BlockFlyoutInflater implements IFlyoutInflater {
+  protected permanentlyDisabledBlocks = new Set<BlockSvg>();
+  protected listeners = new Map<string, browserEvents.Data[]>();
+  protected flyoutWorkspace?: WorkspaceSvg;
+  protected flyout?: IFlyout;
+  private capacityFilter: (e: AbstractEvent) => void;
+
+  constructor() {
+    this.capacityFilter = (event: AbstractEvent) => {
```

Comment:
Why is the capacity filter defined in the constructor instead of just another method?

Reply:
Because we need `this` bound, but updated it to use the wrapper pattern used elsewhere thoughout the codebase.